pkg/agent/billing/billing.go

```diff
@@ -130,7 +161,46 @@ func RunBillingMetricsCollector(
 	}
 }
 
-func (s *metricsState) collect(logger *zap.Logger, store VMStoreForNode, metrics PromMetrics) {
+func collectMetricsForVM(ctx context.Context, logger *zap.Logger, vm *vmapi.VirtualMachine, metricsChan chan vmMetricsKV, promMetrics *PromMetrics) {
+	byteCounts, err := vm.GetNetworkUsage(ctx, 500*time.Millisecond)
```

Comment:
If the VM doesn't have the relevant flag enabled, let's just skip trying to fetch network usage?

Reply:
Added